modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/di/ExperimentModule.kt

```diff
@@ -0,0 +1,68 @@
+package au.com.shiftyjelly.pocketcasts.analytics.di
+
+import android.content.Context
+import au.com.shiftyjelly.pocketcasts.analytics.AccountStatusInfo
+import au.com.shiftyjelly.pocketcasts.analytics.experiments.Experiment
+import au.com.shiftyjelly.pocketcasts.analytics.experiments.ExperimentProvider
+import au.com.shiftyjelly.pocketcasts.servers.di.Cached
+import au.com.shiftyjelly.pocketcasts.utils.log.LogBuffer
+import com.automattic.android.experimentation.ExperimentLogger
+import com.automattic.android.experimentation.VariationsRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import java.io.File
+import javax.inject.Singleton
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import okhttp3.OkHttpClient
+
+@Module
+@InstallIn(SingletonComponent::class)
+object ExperimentModule {
+
+    @Provides
+    @Singleton
+    fun provideExperimentProvider(
+        accountStatusInfo: AccountStatusInfo,
+        repository: VariationsRepository,
+    ): ExperimentProvider = ExperimentProvider(accountStatusInfo, repository)
+
+    @Provides
+    @Singleton
+    fun provideVariationsRepository(
+        @ApplicationContext context: Context,
+        @Cached okHttpClient: OkHttpClient,
+    ): VariationsRepository {
+        val cacheDir = File(context.cacheDir, "experiments_cache").apply {
+            if (!exists()) mkdirs()
+        }
+
+        val experiments = Experiment.getAllExperiments().map { experiment ->
+            com.automattic.android.experimentation.Experiment(experiment.identifier)
+        }.toSet()
+
+        val logger = object : ExperimentLogger {
+            override fun d(message: String) {
+                LogBuffer.i(ExperimentProvider.TAG, message)
+            }
+
+            override fun e(message: String, throwable: Throwable?) {
+                throwable?.let { LogBuffer.e(ExperimentProvider.TAG, throwable, message) } ?: LogBuffer.e(ExperimentProvider.TAG, message)
+            }
+        }
+
+        return VariationsRepository.create(
+            platform = ExperimentProvider.PLATFORM,
+            experiments = experiments,
+            logger = logger,
+            failFast = true,
```

Comment:
```suggestion
            failFast = BuildConfig.DEBUG,
```

Can we please fail fast only in debug? This was intended like this - I think we shouldn't crash the app for users if a code related to experiments is failing.

Reply:
Nice catch! I will update to this